File: spring-boot/grpc/client/src/main/java/no/acntech/sandbox/service/GreetingService.java
package no.acntech.sandbox.service;

import net.devh.boot.grpc.client.inject.GrpcClient;
import org.springframework.stereotype.Service;

import no.acntech.sandbox.model.FormData;
import no.acntech.sandbox.model.GreetingDto;
import no.acntech.sandbox.model.GreetingGrpc;
import no.acntech.sandbox.model.HelloRequest;

@Service
public class GreetingService {

    @GrpcClient("grpc-server")
    private GreetingGrpc.GreetingBlockingStub greetingBlockingStub;

    public GreetingDto getGreeting(FormData formData) {
        var request = HelloRequest.newBuilder()
                .setName(formData.getName())
                .build();
        var response = greetingBlockingStub.sayHello(request);
        return new GreetingDto(response.getMessage());
    }
}

File: spring-boot-kotlin/grpc/client/src/main/kotlin/no/acntech/sandbox/service/GreetingService.kt
package no.acntech.sandbox.service

import net.devh.boot.grpc.client.inject.GrpcClient
import no.acntech.sandbox.model.Greeting
import no.acntech.sandbox.model.GreetingGrpc
import no.acntech.sandbox.model.Hello
import no.acntech.sandbox.model.HelloRequest
import org.springframework.stereotype.Service

@Service
class GreetingService {

    @GrpcClient("grpc-server")
    lateinit var greetingBlockingStub: GreetingGrpc.GreetingBlockingStub

    fun getGreeting(hello: Hello): Greeting {
        val request = HelloRequest.newBuilder()
                .setName(hello.name)
                .build()
        val response = greetingBlockingStub.sayHello(request)
        return Greeting(response.message)
    }
}